tools/pq/backend/postgres.go

```diff
@@ -0,0 +1,100 @@
+package backend
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+
+	"github.com/ThreeDotsLabs/watermill/components/delay"
+	"github.com/ThreeDotsLabs/watermill/tools/pq/cli"
+)
+
+type PostgresMessage struct {
+	Offset   int    `db:"offset"`
+	UUID     string `db:"uuid"`
+	Payload  string `db:"payload"`
+	Metadata string `db:"metadata"`
+}
+
+type PostgresBackend struct {
+	db     *sqlx.DB
+	config cli.BackendConfig
+}
+
+func NewPostgresBackend(ctx context.Context, config cli.BackendConfig) (*PostgresBackend, error) {
```

Comment:
food for thought: how to move it to Postgres adapter potentially 🤔

Reply:
As discussed, let's keep it as is for now, and see if there's a better abstraction in the future.